src/components/DiagramFrame/DiagramFrame.tsx

```diff
@@ -0,0 +1,267 @@
+import { useCurrentModel } from "../../utils/routes"
+import { DiagrammedModel, DiagramMetadata } from "../../utils/diagrammer"
+import "./styles.css"
+import {MetadataPanel} from "./MetadataPanel/MetadataPanel"
+import { DiagramSettings } from "./DiagramSettings"
+import { HelpPanel } from "./HelpPanel"
+import { ViewOptions } from "./ViewOptions"
+import { DiagramHeader } from "./DiagramHeader"
+import { DiagramCanvas } from "./DiagramCanvas/DiagramCanvas"
+import { 
+  ZOOM_INIT,
+  X_INIT,
+  Y_INIT,
+  OVERRIDE_KEY,
+  OVERRIDE_KEY_SUBTLE
+} from '../../utils/constants'
+import { ILookmlModel, ILookmlModelExplore, IGitBranch } from "@looker/sdk/lib/sdk/4.0/models"
+import {DiagramFrameProps, ExploreDropdown, Rail, Stage} from "./types"
+import {getBranchOptions} from "./utils"
+
+export const DiagramFrame: React.FC<DiagramFrameProps> = ({
+  unfilteredModels,
+  pathModelName,
+  pathExploreName,
+  modelDetail,
+  dimensions,
+  modelError,
+  setModelError,
+  hiddenToggle,
+  setHiddenToggle,
+  displayFieldType,
+  setDisplayFieldType,
+  selectedBranch,
+  setSelectedBranch,
+  }) => {
+  const [viewVisible, setViewVisible] = React.useState<VisibleViewLookup>({})
+  const [showSettings, setShowSettings] = React.useState(true)
+  const [showHelp, setShowHelp] = React.useState(false)
+  const [showViewOptions, setShowViewOptions] = React.useState(false)
+  const [showNoAside, setShowNoAside] = React.useState(false)
+  const [reload, setReload] = React.useState(false)
+  const [selectionInfo, setSelectionInfo] = React.useState<SelectionInfoPacket>({})
+  const [zoomFactor, setZoomFactor] = React.useState(ZOOM_INIT)
+  const [viewPosition, setViewPosition] = React.useState({x: X_INIT, y: Y_INIT})
+  const [minimapEnabled, setMinimapEnabled] = React.useState(false)
+  const [minimapUntoggled, setMinimapUntoggled] = React.useState(true)
+  const currentModel = useCurrentModel(selectedBranch, modelError)
+
+  const handleHiddenToggle = (event: any) => setHiddenToggle(event.target.checked)
+
+  function showDiagram() {
+    setShowHelp(false)
+    setShowViewOptions(false)
+    setShowSettings(false)
+    setShowNoAside(true)
+    return true
+  }
+
+  function toggleExploreInfo() {
+    if (currentExplore && modelDetail) {
+      Object.keys(selectionInfo).length === 0 || selectionInfo.lookmlElement !== "explore"
+      ? setSelectionInfo({
+        lookmlElement: "explore"
+      })
+      : setSelectionInfo({})
+    }
+  }
+
+  let modelDetails = unfilteredModels ? unfilteredModels.filter((d: ILookmlModel)=>{ 
+    return d.explores.length >= 1
+  }).map((d: ILookmlModel)=>{
+    return {
+      value: d.name,
+      label: d.label
+    }
+  }).sort((a: SelectOptionProps, b: SelectOptionProps) => a.label < b.label ? -1 : 1) : []
+
+  let currentExplore: ILookmlModelExplore = modelDetail && modelDetail.explores.filter((d: ILookmlModelExplore)=>{
+    return d.name === pathExploreName
+  })[0]
```

Comment:
Optional chaining + implicit return cleans this up some. There are quite a few places in here you could do this if you were so inclined.

```suggestion
  let currentExplore: ILookmlModelExplore = modelDetail?.explores.filter((d: ILookmlModelExplore)=> d.name === pathExploreName)[0]
```